addons/l10n_in/models/company.py

```diff
@@ -11,4 +12,22 @@ class ResCompany(models.Model):
             ("8", "8 Digits"),
         ],
         string="HSN Code Digit",
+        compute="_compute_l10n_in_hsn_code_digit",
+        store=True,
+        readonly=False,
     )
+
+    @api.constrains('l10n_in_hsn_code_digit')
+    def _check_hsn_digit_validation(self):
+        for record in self:
```

Comment:
We could also say that if not set, it won't do any validation.  (that would mean that they do not want to see the banner) (just an idea)